format/postgres/flavours/pgpro12/pg_control.go

```diff
@@ -0,0 +1,220 @@
+/* XXX  4-byte hole */
+/*  152      |     8 */ // XLogRecPtr backupStartPoint;
+/*  160      |     8 */ // XLogRecPtr backupEndPoint;
+/*  168      |     1 */ // _Bool backupEndRequired;
+/* XXX  3-byte hole */
+/*  172      |     4 */ // int wal_level;
+/*  176      |     1 */ // _Bool wal_log_hints;
+/* XXX  3-byte hole */
+/*  180      |     4 */ // int MaxConnections;
+/*  184      |     4 */ // int max_worker_processes;
+/*  188      |     4 */ // int max_wal_senders;
+/*  192      |     4 */ // int max_prepared_xacts;
+/*  196      |     4 */ // int max_locks_per_xact;
+/*  200      |     1 */ // _Bool track_commit_timestamp;
+/* XXX  3-byte hole */
+/*  204      |     4 */ // uint32 maxAlign;
+/*  208      |     8 */ // double floatFormat;
+/*  216      |     4 */ // uint32 blcksz;
+/*  220      |     4 */ // uint32 relseg_size;
+/*  224      |     4 */ // uint32 xlog_blcksz;
+/*  228      |     4 */ // uint32 xlog_seg_size;
+/*  232      |     4 */ // uint32 nameDataLen;
+/*  236      |     4 */ // uint32 indexMaxKeys;
+/*  240      |     4 */ // uint32 toast_max_chunk_size;
+/*  244      |     4 */ // uint32 loblksize;
+/*  248      |     1 */ // _Bool float4ByVal;
+/*  249      |     1 */ // _Bool float8ByVal;
+/* XXX  2-byte hole */
+/*  252      |     4 */ // uint32 data_checksum_version;
+/*  256      |    32 */ // char mock_authentication_nonce[32];
+/*  288      |     4 */ // pg_icu_version icu_version;
+/*  292      |     4 */ // pg_crc32c crc;
+//
+/* total size (bytes):  296 */
+
+// type = struct CheckPoint {
+/*    0      |     8 */ // XLogRecPtr redo;
+/*    8      |     4 */ // TimeLineID ThisTimeLineID;
+/*   12      |     4 */ // TimeLineID PrevTimeLineID;
+/*   16      |     1 */ // _Bool fullPageWrites;
+/* XXX  7-byte hole */
+/*   24      |     8 */ // FullTransactionId nextFullXid;
+/*   32      |     4 */ // Oid nextOid;
+/*   36      |     4 */ // MultiXactId nextMulti;
+/*   40      |     4 */ // MultiXactOffset nextMultiOffset;
+/*   44      |     4 */ // TransactionId oldestXid;
+/*   48      |     4 */ // Oid oldestXidDB;
+/*   52      |     4 */ // MultiXactId oldestMulti;
+/*   56      |     4 */ // Oid oldestMultiDB;
+/* XXX  4-byte hole */
+/*   64      |     8 */ // pg_time_t time;
+/*   72      |     4 */ // TransactionId oldestCommitTsXid;
+/*   76      |     4 */ // TransactionId newestCommitTsXid;
+/*   80      |     4 */ // TransactionId oldestActiveXid;
+/* XXX  4-byte padding */
+//
+/* total size (bytes):   88 */
+
+func DecodePgControl(d *decode.D, in any) any {
+	d.SeekAbs(0)
+	/*    0      |     8 */ // uint64 system_identifier;
+	/*    8      |     4 */ // uint32 pg_control_version;
+	/*   12      |     4 */ // uint32 catalog_version_no;
+	/*   16      |     4 */ // DBState state;
+	/* XXX  4-byte hole  */
+	d.FieldU64("system_identifier")
+	d.FieldU32("pg_control_version", common.VersionMapper)
+	d.FieldU32("catalog_version_no")
+	d.FieldU32("state", common.DBState)
+	d.U32()
```

Comment:
Add a fields for the holes? `hole0`, `hole1`, ..? otherwise fill gaps will add unknown for them. That way the bit range will be known to be a hole on purpose